jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/AbstractConnection.java

```diff
@@ -32,7 +32,7 @@
  * will schedule a callback to {@link #onFillable()} or {@link #onFillInterestedFailed(Throwable)}
  * as appropriate.</p>
  */
-public abstract class AbstractConnection implements Connection
+public abstract class AbstractConnection implements Connection, Invocable
```

Comment:
Why does `AbstractConnection` needs to be `Invocable`? 
We should rely on `Callback`s to be `Invocable`, the connection itself seems too coarse of an object.

Reply:
XyzConnections use the plain `fillInterested()` registration which uses the `ReadCallback` in AbstractConnection.  Previously that was always BLOCKING.  We now need it to be Invocable if we want to get non blocking dispatch for the very first part of the request.

Alternately, I guess we could make HttpConnection not use `fillInterested()` and instead use `tryFillInterested(Callback)`, but then h2 and h3 would have to do the same.